x-pack/plugin/security/src/main/java/org/elasticsearch/xpack/security/authz/IndicesAndAliasesResolver.java

```diff
@@ -360,12 +364,27 @@ ResolvedIndices resolveIndicesAndAliases(
                             nameExpressionResolver,
                             indicesRequest.includeDataStreams()
                         )) {
-                            resolvedIndicesBuilder.addLocal(
+                            localExpressions.add(
                                 IndexNameExpressionResolver.combineSelectorExpression(authorizedIndex, allIndicesPatternSelector)
                             );
                         }
                     }
                 }
+
+                var resolvedExpressionsBuilder = ResolvedIndexExpressions.builder();
+                resolvedExpressionsBuilder.addExpressions(
+                    Metadata.ALL,
+                    localExpressions,
+                    ResolvedIndexExpression.LocalIndexResolutionResult.SUCCESS,
+                    Collections.emptySet()
```

Comment:
This needs to be implemented by expanding CPS remote indices similar to how it is done in the else branch https://github.com/elastic/elasticsearch/blob/c468e3cd87a6ea32ed32b1f99e17131f0e52ca8e/x-pack/plugin/security/src/main/java/org/elasticsearch/xpack/security/authz/IndicesAndAliasesResolver.java#L373-L390